libexec/bats-core/bats-exec-file

```diff
@@ -157,25 +158,27 @@ bats_run_tests() {
     test_pids_and_numbers=()
     output_folder="$BATS_RUN_TMPDIR/parallel_output"
     parallel_log_file="$BATS_RUN_TMPDIR/parallel.log"
+    test_number_in_file=0
     for test_name in "${tests_to_run[@]}"; do
       # Only handle non-empty lines
       if [[ $test_name ]]; then
-        ((++test_number))
-        test_output_dir="$output_folder/$test_number"
+        ((++test_number_in_run))
+        ((++test_number_in_file))
```

Comment:
These should have similar sounding names to the BATS variables (`in_run` sounds odd and I got confused when reading the later part of the patch). Maybe `test_number_in_file` and just `test_number_global`.